src/Asset/Capacity/HasOperatingSystemCapacity.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset\Capacity;
+
+use CommonGLPI;
+use Item_OperatingSystem;
+use OperatingSystem;
+
+/**
+ * Operating system capacity for assets composition.
+ *
+ * When enabled:
+ * - Register the asset in $CFG_GLPI["operatingsystem_types"].
+ * - The "Item_OperatingSystem system" tab will be added to the asset.
+ * - The "Item_OperatingSystem" search options will be added to the asset.
+ *
+ * When disabled:
+ * - Unregister the asset in $CFG_GLPI["operatingsystem_types"].
+ * - All "Item_OperatingSystem" items linked to the asset will be deleted.
+ * - All references to "Item_OperatingSystem" and "OperatingSystem" in the asset
+ * history will be deleted.
+ * - All display preferences related to "Item_OperatingSystem" for the asset
+ * will be deleted.
+ */
```

Comment:
I bet this comment will not be maintained and will be outdated as long as we will start to implement transferse features, like clone, transfer, templates, ...

The capacity name is already self-describing.

```suggestion
```